### PR TITLE
Reth v1.1.1

### PR DESCRIPTION
**Pull request by ZanCorDX**
## 📝 Summary

- Dependency hell.			
- In lots of generics (eg:ProviderFactory) the need of DB+Spec was replaced for NodeTypesWithDB.
- DatabaseProviderFactory DB now is asociated type and have 2 more. In lots of places now we need Provider: BlockReader.
- Lots of types moved to alloy_primitives.
- encode/decode_enveloped now are XXX_2718 with minimal changes.
- New alloy TrieNode::EmptyRoot .
- CachedReads moved to reth::revm::cached.
- AsyncStateRoot replaced by ParallelStateRoot (https://github.com/paradigmxyz/reth/pull/11213).
- Some HashMap/Set changed to custom versions (eg: alloy_primitives::map::HashMap).
- TransactionPool needs stronger restrictions on Transaction.
- Updated block finalization (pectra).
- Pectra relay submition.

## 💡 Motivation and Context

- We needed this for Pectra and to be up to date with reth.
- My life was going so well that I needed something to balance it so I did this boring PR.

## ✅ I have completed the following steps:

* [X] Run `make lint`
* [X] Run `make test`
* [ ] Added tests (if applicable)

### CODE REVIEW

File: crates/eth-sparse-mpt/src/sparse_mpt/fixed_trie.rs [hunk 1]
```diff
@@ -111,7 +106,8 @@ impl From<AlloyBranchNode> for FixedBranchNode {
                 let rlp_data = stack_iter
                     .next()
                     .expect("stack must be the same size as mask");
-                children[index as usize] = Some(rlp_data.into());
+                // @Pending: Eval replacing Bytes for ArrayVec to avoid the copy or dig deeper into low-level.
+                children[index as usize] = Some(Bytes::copy_from_slice(rlp_data.as_ref()));
```
Comment (liamaharon) [inline crates/eth-sparse-mpt/src/sparse_mpt/fixed_trie.rs:110]: Can we bench it?

File: crates/rbuilder/src/building/builders/parallel_builder/simulation_cache.rs [hunk 2]
```diff
@@ -208,11 +208,13 @@ mod tests {
 
         fn create_cached_simulation_state(&mut self) -> CachedSimulationState {
             let mut cached_reads = CachedReads::default();
-            let mut storage = HashMap::new();
+            //let mut storage = HashMap::new();
```
Comment (liamaharon) [inline crates/rbuilder/src/building/builders/parallel_builder/simulation_cache.rs]:
```suggestion
```

File: crates/rbuilder/src/mev_boost/sign_payload.rs [hunk 3]
```diff
@@ -148,11 +151,17 @@ pub fn sign_block_for_relay(
...
             transactions: sealed_block
                 .body
+                .transactions
                 .iter()
-                .map(|tx| tx.envelope_encoded().to_vec().into())
+                .map(|tx| {
+                    let mut buf = Vec::new();
+                    tx.encode_2718(&mut buf);
+                    buf.into()
+                })
```
Comment (liamaharon) [inline crates/rbuilder/src/mev_boost/sign_payload.rs:160]: Do we have a way to ensure the new encoding is correct?

File: crates/rbuilder/src/primitives/fmt.rs [hunk 4]
```diff
@@ -14,7 +14,7 @@ pub fn write_share_bundle_tx<Buffer: Write>(
         "TX {} Rev  {:?} val {}\n",
         tx.tx.hash(),
         tx.revert_behavior,
-        tx.tx.tx.value()
+        tx.tx.value()
```
Comment (liamaharon) [inline crates/rbuilder/src/primitives/fmt.rs:17]: 😁

File: crates/transaction-pool-bundle-ext/Cargo.toml [hunk 5]
```diff
@@ -7,8 +7,14 @@ edition.workspace = true
 reth = { workspace = true }
 reth-eth-wire-types = { workspace = true }
 reth-primitives = { workspace = true }
-reth-rpc-types = { workspace = true }
+#reth-rpc-types = { workspace = true }
```
Comment (liamaharon) [inline crates/transaction-pool-bundle-ext/Cargo.toml]:
```suggestion
```

File: crates/transaction-pool-bundle-ext/bundle_pool_ops/rbuilder/Cargo.toml [hunk 6]
```diff
@@ -7,10 +7,12 @@ edition = "2021"
 transaction-pool-bundle-ext = { path = "../.." }
 rbuilder = { path = "../../../rbuilder" }
 
+alloy-primitives.workspace = true
+alloy-rpc-types-beacon.workspace = true
 reth-primitives = { workspace = true }
 reth-provider = { workspace = true }
 reth-db-api = { workspace = true }
-reth-rpc-types = { workspace = true }
+#reth-rpc-types = { workspace = true }
```
Comment (liamaharon) [inline crates/transaction-pool-bundle-ext/bundle_pool_ops/rbuilder/Cargo.toml]:
```suggestion
```

**Review by liamaharon — APPROVED** [review]
LGTM after comments if we're happy with its staging performance.

**Review by ferranbt — APPROVED** [review]

File: crates/rbuilder/src/mev_boost/sign_payload.rs (hunk 3 above)
Comment (ZanCorDX) [reply crates/rbuilder/src/mev_boost/sign_payload.rs:160]: Not that I remember. We trust on reth tests.

File: crates/eth-sparse-mpt/src/sparse_mpt/fixed_trie.rs (hunk 1 above)
Comment (ZanCorDX) [reply crates/eth-sparse-mpt/src/sparse_mpt/fixed_trie.rs:110]: This is for @dvush

**Pull request merged**